### PR TITLE
compute all descendant levels of a script

### PR DESCRIPTION
**Pull request by davidsbailey**
## Description

As part of wanting to clean up unused scripts and levels, we want to be able to see what levels are depended upon by each script. This method will also serve as a helper method in computing whether a level should have its instructions turned into speech via TTS. Today, existing helper methods which compute the levels in a script do not include:
* contained levels
* project-template-backed levels
* bubble choice sublevels
* level group sublevels

This PR adds a helper method which includes those types of dependent/descendant levels in its return value.

## Testing story

New unit test covers new behavior. The test's implementation is a bit too similar to the method's implementation for my taste, but that will change over time as we move more child level types onto the new parent_levels / child_levels associations in the level object.

# Reviewer Checklist:

- [ ] Tests provide adequate coverage
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

**Review by Hamms — COMMENTED** [review]
I don't love the extent to which this puts level-implementation-specific logic into the Script model. I'd also point out that although I don't know of any scenarios in which we nest these various forms of descendant levels I also don't know of anything that would prevent us from doing so in the future.

What if instead we implemented this as a helper method on the level model, which can grab all descendant levels for a given level. We can then implement the specific methods for getting that information directly on the appropriate level time, and easily recurse.

File: dashboard/app/models/levels/bubble_choice.rb [hunk 1]
```diff
@@ -63,6 +63,13 @@ def sublevel_position(sublevel)
     i.present? ? i + 1 : nil
   end
 
+  # no need to call super or recurse here, because bubble choice levels and
+  # their children cannot have contained levels or project template levels.
```
Comment (Hamms) [inline dashboard/app/models/levels/bubble_choice.rb]: just to satisfy my own curiosity; what is it that prevents the child levels from having their own child levels, both here and in level groups?

**Review by Hamms — APPROVED** [review]

File: dashboard/app/models/levels/bubble_choice.rb (hunk 1 above)
Comment (davidsbailey) [reply dashboard/app/models/levels/bubble_choice.rb]: turns out I was mistaken here. I am updating this to take a recursive approach and will tag you again when it is ready for another look.

**Comment by davidsbailey** [comment]
@Hamms PTAL! In addition to added tests, I validated this change by running `rake seed:dsls`, which should catch any problems seeding bubble choice or level group levels.

**Comment by davidsbailey** [comment]
one more tweak -- no_contained_or_template_levels is not needed because contained levels and project template levels can only exist on Blockly levels, whereas BubbleChoice and LevelGroup are DSLDefined levels.

**Pull request merged**